backend/capellacollab/settings/modelsources/t4c/repositories/injectables.py

```diff
@@ -0,0 +1,51 @@
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import typing as t
+
+from fastapi import Depends, HTTPException
+from sqlalchemy.exc import NoResultFound
+from sqlalchemy.orm import Session
+
+from capellacollab.core.database import get_db
+from capellacollab.settings.modelsources.t4c.injectables import (
+    get_existing_instance,
+)
+from capellacollab.settings.modelsources.t4c.models import DatabaseT4CInstance
+from capellacollab.settings.modelsources.t4c.repositories import crud
+from capellacollab.settings.modelsources.t4c.repositories.models import (
+    DatabaseT4CRepository,
+)
+
+
+def get_existing_instance_repository(
+    t4c_repository_id: t.Optional[int] = None,
+    db: Session = Depends(get_db),
+    instance: t.Optional[DatabaseT4CInstance] = Depends(get_existing_instance),
+) -> t.Optional[DatabaseT4CRepository]:
```

Comment:
Similar to the `get_existing_instance`, we can set the return type here to `DatabaseT4CRepository`, so we do not have to handle the `None` cases at places where we use this injectable.

Reply:
Same as above.